pandas/tests/io/generate_legacy_storage_files.py

```diff
@@ -53,10 +53,7 @@
     Period,
     RangeIndex,
     Series,
-    SparseDataFrame,
```

Comment:
I think you can leave this file intact for now? 
(isn't it meant to be run with older version of pandas? In case we need to regenerate old pickle files, we might still want to generate sparse ones to check that reading them works / errors with a good message)

Reply:
IIUC, we need to remove it so that the sparse structures are not included, re-run it with an older pandas, and commit the new binary file.

I will update to use SparseArray though.

Reply:
@WillAyd can you comment here? I think we import this file when running the tests, so we need to be able to import it under both the old and new pandas?

Reply:
Should this file now be restored now you added the pickle compat? (I also think you now no longer update the old pickle files, right?)

Reply:
I think this file gets imported on master, so I need to remove this.

Reply:
if so, you can put it behind a try/except?

Reply:
That alone probably isn't what we want though, right? Since even if it's in a try / except, it's not actually being used.

I think the best thing is to include newly generated legacy files, in addition to the old ones. Then we verify that the new legacy files (with Series[sparse] rather than SparseSeries, etc.) are working, and that we're still able to read old files generated from SparseSeries / SparseDataFrame.

Reply:
I am not very familiar with how this is used in practice, but I thought it is something like:

- we decide to raise minimum supported versions on read from 0.20 to 0.23 (we just bumped to minimum of 0.20 in 0.25)
- we run this script with pandas 0.23 to generated the new "legacy pickle files" and replace the old ones with those

If that is correct (@WillAyd is that what you did for 0.25?), then the script still needs to be able to generate pickles files with SparseDataFrame (in this hypothetical case, if we would directly bump to eg 1.0, we wouldn't need it)

Reply:
Sorry I missed the original ping. I haven't been able to check in detail but will carve out some time later today. If it helps in the interim here is where I cleaned up the old compat files for Panel:

https://github.com/pandas-dev/pandas/pull/27082

Reply:
The changes here are correct; you just need to re-run this in an environment with pandas 0.20.3 ideally using the minimum supported Python version, which I guess right now is 3.5.3 though by the time this goes live should be 3.6, so maybe stick to the latter (I see there is one for 3.5.6 already)

Reply:
@WillAyd can you answer on my comment above in this thread? (https://github.com/pandas-dev/pandas/pull/28425/files#r324859383) Why would you not want to recreate the pickle files *with* SparseDataFrame? Those legacy pickle files are meant to represent the legacy situation, no? And in 0.20.3, those classes exist, and can be written to pickle files.

Reply:
I don’t know the entire history of these files either so may be missing intent, but if we don’t have a SparseDataFrame anymore then there wouldn’t be a reason to try and read one as that object can’t be recreated on master, no?

Reply:
But this file is not mean to recreate them with master, but with an older release? (or is that the part I am misunderstanding? You ran that script with pandas 0.21, not with master?)

Reply:
Right the file would be created with an old release but read and tested against master I think

Reply:
So if we want to mimic what users can do and what Tom is trying to support (i.e. write a pickle file with SparseDataFrame in an older release and read this file with the new release), this file needs to be able to generate such pickle files with SparseDataFrame?

Now, I don't really care that much about SparseDataFrame, and there are currently already legacy pickle files that contain it, so the need to be able to recreate those files before we will actually remove that support is maybe not that high, and I am maybe splitting hairs.

Reply:
Sorry @jorisvandenbossche, I think I've misunderstood you this whole time (and I was confused why you kept pushing on this point 😄). You're correct that if we relied on these legacy files for testing reading files written with SparseSeries & SparseDataFrame, then we'd need to keep the imports here (at conditionally).

But https://github.com/pandas-dev/pandas/pull/28425/commits/a8b0d65154b76fcb0b62025d87d8ead9fcdde723 added tests dedicated for that (we need dedicated tests, since the existing ones silence all warnings).

So from a test coverage POV, we're OK with removing writing SparseSeries & SparseDataFrame here. It's just a matter of code style / prep for running this file when 1.0.0 becomes the "legacy" pickle version we support.

Reply:
> So from a test coverage POV, we're OK with removing writing SparseSeries & SparseDataFrame here.

Yep, if we have files now, it seems ok to remove it from this file. It seems rather unlikely that after the next bump in legacy support for pickle files, we still want to keep support for SparseDataFrame, I suppose, as we now warn for it?

Reply:
That sounds right to me. Technically, we could chose to bump the next legacy to version 0.25.x, which has SparseDataFrame, but I think we're more likely to choose 1.0.x since it's a round number.